e2e_playwright/st_audio_input_test.py

```diff
@@ -0,0 +1,138 @@
+
+
+def test_snapshots(themed_app: Page, assert_snapshot: ImageCompareFunction):
+    audio_input_element = themed_app.get_by_test_id("stAudioInput").first
+
+    assert_snapshot(audio_input_element, name="st_audio_input_default")
+
+
+@pytest.mark.only_browser("chromium")
+def test_audio_input_works_in_forms(app: Page):
+    app.context.grant_permissions(["microphone"])
+
+    expect(app.get_by_text("Audio Input in Form: None")).to_be_visible()
+
+    form_audio_input = app.get_by_test_id("stAudioInput").nth(1)
+    form_audio_input.get_by_role("button", name="Record").click()
+    time.sleep(1)
+    form_audio_input.get_by_role("button", name="Stop recording").click()
+
+    expect(app.get_by_text("Audio Input in Form: None")).to_be_visible()
+
+    click_form_button(app, "Submit")
+    wait_for_app_run(app)
+
+    expect(app.get_by_text("Audio Input in Form: None")).not_to_be_visible()
+
+
+@pytest.mark.only_browser("chromium")
+def test_feedback_works_with_fragments(app: Page):
+    app.context.grant_permissions(["microphone"])
+
+    expect(app.get_by_text("Runs: 1")).to_be_visible()
+    expect(app.get_by_text("Audio Input in Fragment: None")).to_be_visible()
+
+    fragment_audio_input = app.get_by_test_id("stAudioInput").nth(2)
+    fragment_audio_input.get_by_role("button", name="Record").click()
+    time.sleep(1)
+    fragment_audio_input.get_by_role("button", name="Stop recording").click()
+    wait_for_app_run(app)
+
+    expect(app.get_by_text("Audio Input in Fragment: None")).not_to_be_visible()
+    expect(app.get_by_text("Runs: 1")).to_be_visible()
+
+    fragment_audio_input.get_by_role("button", name="Clear recording").click()
+    wait_for_app_run(app)
+
+    expect(app.get_by_text("Runs: 1")).to_be_visible()
+
+
+@pytest.mark.only_browser("chromium")
+def test_audio_input_basic_flow(app: Page):
+    app.context.grant_permissions(["microphone"])
+
+    audio_input = app.get_by_test_id("stAudioInput").first
+
+    expect(
+        app.get_by_text("This app would like to use your microphone.").first
+    ).not_to_be_visible()
+
+    record_button = app.get_by_role("button", name="Record").first
+    clock = audio_input.get_by_test_id("stAudioInputWaveformTimeCode")
+
+    expect(clock).to_have_text("00:00")
+
+    record_button.click()
+
+    stop_button = audio_input.get_by_role("button", name="Stop recording").first
+    expect(stop_button).to_be_visible()
+
+    time.sleep(2)
+
+    stop_button.click()
+
+    play_button = audio_input.get_by_role("button", name="Play").first
+
+    expect(clock).not_to_have_text("00:00")
+
+    play_button.click()
+
+    pause_button = audio_input.get_by_role("button", name="Pause").first
+    expect(pause_button).to_be_visible()
+
+    pause_button.click()
+
+    expect(play_button).to_be_visible()
+
+    audio_input.hover()
+
+    clear_button = audio_input.get_by_role("button", name="Clear recording").first
+    expect(clear_button).to_be_visible()
+
+    clear_button.click()
```

Comment:
Clearing should also trigger a rerun and clear the value on the backend? Thats something that could also be verified here.